egs2/TEMPLATE/sds1/app.py

```diff
@@ -0,0 +1,797 @@
+
+from espnet2.sds.espnet_model import ESPnetSDSModelInterface
+
+access_token = os.environ.get("HF_TOKEN")
+ASR_name = None
+LLM_name = None
+TTS_name = None
+ASR_options = []
+LLM_options = []
+TTS_options = []
+Eval_options = []
+upload_to_hub = None
+dialogue_model = None
+
+
+def read_args():
+    global access_token
+    global ASR_name
+    global LLM_name
+    global TTS_name
+    global ASR_options
+    global LLM_options
+    global TTS_options
+    global Eval_options
+    global upload_to_hub
+    global dialogue_model
+    parser = argparse.ArgumentParser(
+        description="Run the app with HF_TOKEN as a command-line argument."
+    )
+    parser.add_argument(
+        "--HF_TOKEN", required=True, help="Provide the Hugging Face token."
+    )
+    parser.add_argument(
+        "--asr_options",
+        required=True,
+        help="Provide the possible ASR options available to user.",
+    )
+    parser.add_argument(
+        "--llm_options",
+        required=True,
+        help="Provide the possible LLM options available to user.",
+    )
+    parser.add_argument(
+        "--tts_options",
+        required=True,
+        help="Provide the possible TTS options available to user.",
+    )
+    parser.add_argument(
+        "--eval_options",
+        required=True,
+        help="Provide the possible automatic evaluation metrics available to user.",
+    )
+    parser.add_argument(
+        "--default_asr_model",
+        required=False,
+        default="pyf98/owsm_ctc_v3.1_1B",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--default_llm_model",
+        required=False,
+        default="meta-llama/Llama-3.2-1B-Instruct",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--default_tts_model",
+        required=False,
+        default="kan-bayashi/ljspeech_vits",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--upload_to_hub",
+        required=False,
+        default=None,
+        help="Hugging Face dataset to upload user data",
+    )
+    args = parser.parse_args()
+    access_token = args.HF_TOKEN
+    ASR_name = args.default_asr_model
+    LLM_name = args.default_llm_model
+    TTS_name = args.default_tts_model
+    ASR_options = args.asr_options.split(",")
+    LLM_options = args.llm_options.split(",")
+    TTS_options = args.tts_options.split(",")
+    Eval_options = args.eval_options.split(",")
+    upload_to_hub = args.upload_to_hub
+    dialogue_model = ESPnetSDSModelInterface(
+        ASR_name, LLM_name, TTS_name, "Cascaded", access_token
+    )
+
+
+read_args()
+api = HfApi()
+nltk.download("averaged_perceptron_tagger_eng")
+
+latency_ASR = 0.0
+latency_LM = 0.0
+latency_TTS = 0.0
+
+text_str = ""
+asr_output_str = ""
+vad_output = None
+audio_output = None
+audio_output1 = None
+LLM_response_arr = []
+total_response_arr = []
```

Comment:
I think we should define hyper parameters first in this script, then functions, and then run those lines such as `read_args()` or `api=HfApi()`

Reply:
Sounds good! I reordered the script to first define hyperparmaters, then the function followed by the Executable Script